### PR TITLE
OpenGov in Westend and Rococo

### PR DESCRIPTION
**Pull request by al3mart**
Migrating [PR from the archived polkadot repo](https://github.com/paritytech/polkadot/pull/7272)

As per https://github.com/paritytech/polkadot/pull/7272#issuecomment-1559240466, the changes in this MR include the following pallets into [x] Rococo and [x] Westend runtimes:

    pallet_conviction_voting
    pallet_referenda
    pallet_ranked_collective
    pallet_custom_origins
    pallet_whitelist

And only for westend-runtime:

    pallet_treasury

Following [Kusama runtime config](https://github.com/paritytech/polkadot/tree/dbae30efe080a1d41fe54ef4da8af47614c9ca93/runtime/kusama/src) as a baseline.

Benchmarking of the following pallets done for Rococo and Westend:

    pallet_conviction_voting
    pallet_referenda
    pallet_ranked_collective
    pallet_whitelist

And only for Westend:

    pallet_treasury

Removed Gov1 from Rococo as in  https://github.com/paritytech/polkadot/pull/6701

### CODE REVIEW

**Comment by al3mart** [comment]
All the last comments have been addressed and this changes should be ready for a final review

**Comment by gilescope** [comment]
I think this is an important PR for several reasons:

- It enables people to test governance proposals without having to send them to try them first time for real on a financial network.
- It means the runtime code differences between production and test runtimes are minimised meaning that important differences are less likely to be missed.

The rococo assumes a fellowship to allow whitelisting - I think that's good to have the smallest difference, but I don't see a collectives on rococo at the moment so I think one needs to be rolled out there? (again I don't see a problem with this)

**Comment by al3mart** [comment]
@gilescope thanks for the feedback and contributions, I would invite you to leave a review in case you have time for it just for being a bit closer to unlock that merge :)

@muharem @joepetrowski @mordamax I believe this PR is in good shape and we could try to merge it if there is no further changes required

File: polkadot/runtime/rococo/src/governance/tracks.rs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+const TRACKS_DATA: [(u16, pallet_referenda::TrackInfo<Balance, BlockNumber>); 15] = [
+	(
+		0,
+		pallet_referenda::TrackInfo {
+			name: "root",
+			max_deciding: 1,
+			decision_deposit: 100 * GRAND,
+			prepare_period: 2 * HOURS,
+			decision_period: 14 * DAYS,
```
Comment (gilescope) [inline polkadot/runtime/rococo/src/governance/tracks.rs:76]: I would think we would want much faster decision periods for test networks. Maybe a day of voting with little confirm period and little enactment period?

File: polkadot/runtime/westend/constants/src/lib.rs [hunk 2]
```diff
@@ -96,6 +96,26 @@ pub mod fee {
...
+		const ROOT_INDEX: u32 = 0;
+		// The bodies corresponding to the Polkadot OpenGov Origins.
+		pub const FELLOWSHIP_ADMIN_INDEX: u32 = 1;
+	}
+}
+
+/// System Parachains.
+pub mod system_parachain {
+	/// Statemint parachain ID.
+	pub const WESTMINT_ID: u32 = 1000;
```
Comment (gilescope) [inline polkadot/runtime/westend/constants/src/lib.rs:114]: Probably should call this ASSET_HUB_ID

**Review by gilescope — APPROVED** [review]

File: polkadot/runtime/rococo/src/governance/tracks.rs (hunk 1 above)
Comment (mordamax) [reply polkadot/runtime/rococo/src/governance/tracks.rs:76]: yea that's true, it's for testing of features, not real decisions. 
How quick we can possibly make it?

**Comment by al3mart** [comment]
Comments addressed.

I have reduce the periods duration to keep everything in the range of the minutes. Maintaining the differences between tracks and prepare, decision, confirmation and min_enactment periods to still keep some coherence with the spans of time that are on both production networks.

**Review by chevdor — APPROVED** [review]
Great to test but also nice for people to test those features out to learn.

It would be good to ping @devops-parity and the sudos of those chains to ensure they are aware of the risks and that a recovery plan is drafted in case of an annoyance sneaks in to the short voting periods.

**Review by liamaharon — COMMENTED** [review]
While we're at it, we should probably clean up the Gov V1 storage as was done in https://github.com/paritytech/polkadot/pull/7314

**Comment by mordamax** [comment]
@liamaharon @al3mart could this cleanup be done in a separate PR? 
we really need this to be merged asap :) 

Thanks!

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-tracing
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3798101

**Comment by al3mart** [comment]
As a heads up I checked the migrations with try-runtime and got the following result for Rococo runtime
`INFO  try-runtime::cli] Consumed ref_time: 2.053575s (102.68% of max 2s)`

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-3-0/4614/1

**Pull request merged**